File: src/c/ch01/1.3/l0/divisibility.c
#include "divisibility.h"
#include <stdlib.h>

bool is_divisible(int64_t a, int64_t b) {
    /* divisibility by zero is undefined */
    if (b == 0) return false;
    return (a % b) == 0;
}

int last_decimal_digit(int64_t n) {
    int64_t m = llabs(n);
    return (int)(m % 10);
}

bool divisible_by_10(int64_t n) {
    return last_decimal_digit(n) == 0;
}

File: src/python/ch01/1.3/l0/divisibility.py
def is_divisible(a: int, b: int) -> bool:
    """True iff b divides a (b != 0)."""
    if b == 0:
        raise ZeroDivisionError("divisibility by zero undefined")
    return a % b == 0

def last_digit(n: int) -> int:
    """Return the last decimal digit of n."""
    return abs(n) % 10

def divisible_by_10(n: int) -> bool:
    """Quick rule: last digit is 0."""
    return last_digit(n) == 0
